cmd/snap/inhibit.go

```diff
@@ -37,15 +37,26 @@ import (
 )
 
 var runinhibitWaitWhileInhibited = runinhibit.WaitWhileInhibited
+var runinhibitIsLocked = runinhibit.IsLocked
 
 // errSnapRefreshConflict indicates that a retry is needed because snap-run
 // might have started without a hint lock file and now there is an ongoing refresh
 // which could alter the current snap revision.
 var errSnapRefreshConflict = fmt.Errorf("snap refresh conflict detected")
 
 // maybeWaitWhileInhibited is a wrapper for waitWhileInhibited that skips waiting
-// if refresh-app-awareness flag is disabled.
+// if refresh-app-awareness flag is disabled and early-exits when snap run is
+// inhibited for removal.
 func maybeWaitWhileInhibited(ctx context.Context, cli *client.Client, snapName string, appName string) (info *snap.Info, app *snap.AppInfo, hintFlock *osutil.FileLock, err error) {
+	inhibitedForRemove, err := isInhibitedForRemove(snapName)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	// XXX: do we want to allow services to start during removal?
+	if inhibitedForRemove {
+		return nil, nil, nil, fmt.Errorf(i18n.G("snap %q is being removed"), snapName)
```

Comment:
I think I'd add `errInhibitedForRemove` (like we do for errSnapRefreshConflict)  and return it here, let the caller deal with the handling.